frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -106,6 +113,15 @@ public final class EnhancedGraphTable extends BaseEnhancedGraphTable
   @UIEffect
   private @Nullable String selectedBranchName;
 
+  // To accurately track the change of the current branch from the begging, let's put something impossible
+  // as a branch name. This is required to detect the moment when the unmanaged branch notification should be shown.
+  @UIEffect
```

Comment:
Same as below 👇🏻 this is accessed outside of UI thread in `trackCurrentBranchChange`, so pls remove `@UIEffect` and wrap into an `AtomicReference`

Reply:
Or maybe rather... pls remind, `trackCurrentBranchChange` executes on UI thread or outside?

Reply:
it executes on UI thread (stopping there with debug causes UI freeze)